main.rb

```diff
@@ -0,0 +1,141 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    array = to_a
+    i = 0
+    while i < array.length
+      yield(array[i])
+      i += 1
+    end
+    array
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    array = to_a
+    i = 0
+    while i < array.length
+      yield(array[i], i)
+      i += 1
+    end
+    array
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each { |elem| new_array.push(elem) if yield(elem) == true }
+    new_array
+  end
+
+  def my_all?
+    return to_enum(:my_all?) unless block_given?
+
+    my_each { |elem| return false if yield(elem) == false }
+    true
+  end
+
+  def my_any?
+    return to_enum(:my_any?) unless block_given?
+
+    my_each { |elem| return true if yield(elem) == true }
+    false
+  end
+
+  def my_none?
+    return to_enum(:my_none?) unless block_given?
+
+    my_each { |elem| return false if yield(elem) == true }
+    true
+  end
+
+  def my_count(num = nil)
+    counter = 0
+    if num
+      my_each { |elem| counter += 1 if elem == num }
+
+    elsif block_given?
+      my_each { |elem| counter += 1 if yield(elem) }
+
+    else
+      counter = size
+    end
+    counter
+  end
```

Comment:
Enumerable `my_none?` when no block or argument is given returns true only if none of the collection members is true
      Failure/Error: `expect(false_array.my_none?).to be true`

        expected true
             got #<Enumerator:2160> => #<Enumerator: [nil, false, nil, false]:my_none?>
##
Enumerable `my_none?` when no block or argument is given returns false only if one of the collection members is true
      Failure/Error: `expect(true_array.my_none?).to be false`

        expected false
             got #<Enumerator:2180> => #<Enumerator: [nil, false, true, []]:my_none?>
##
Enumerable `my_none?` when a class is passed as an argument returns true if none of the collection is a member of such class
      Failure/Error: `expect(array.my_none?(String)).to be array.my_none?(String)`

      ArgumentError:
        wrong number of arguments (given 1, expected 0)
##
Enumerable `my_none?` when a class is passed as an argument returns true if none of the collection is a member of such class::Numeric
      Failure/Error: `expect(array.my_none?(Numeric)).to be array.my_none?(Numeric)`

      ArgumentError:
        wrong number of arguments (given 1, expected 0)
##
Enumerable `my_none?` when a class is passed as an argument returns false if any of the collection is a member of such class
      Failure/Error: `expect(array.my_none?(String)).to be array.my_none?(String)`

      ArgumentError:
        wrong number of arguments (given 1, expected 0)
##
Enumerable `my_none?` when a Regex is passed as an argument returns true if none of the collection matches the Regex
      Failure/Error: `expect(words.my_none?(/z/)).to be words.none?(/z/)`

      ArgumentError:
        wrong number of arguments (given 1, expected 0)
##
Enumerable `my_none?` when a Regex is passed as an argument returns false if any of the collection matches the Regex
      Failure/Error: `expect(words.my_none?(/d/)).to be words.none?(/d/)`

      ArgumentError:
        wrong number of arguments (given 1, expected 0)
##
Enumerable `my_none?` when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
      Failure/Error: `expect(words.my_none?(5)).to be words.none?(5)`

      ArgumentError:
        wrong number of arguments (given 1, expected 0)
##
Enumerable `my_none?` when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern
      Failure/Error: `expect(words.my_none?(5)).to be words.none?(5)`

      ArgumentError:
        wrong number of arguments (given 1, expected 0)
##